cumulusci/utils/yaml/cumulusci_yml.py

```diff
@@ -104,19 +92,19 @@ class PreflightCheck(CCIDictModel):
 class Plan(CCIDictModel):  # MetaDeploy plans
     title: str
     description: str = None
-    tier: str = None
-    slug: str = None
+    tier: str = Literal["primary", "secondary", "additional"]
+    slug: str
     is_listed: bool = True
-    steps: Dict[str, Union[FlowReference, TaskReference]]
-    checks: List[Check] = []
+    steps: Dict[str, Step]
+    checks: List[PreflightCheck] = []
     group: str = None
     error_message: str = None
     post_install_message: str = None
     preflight_message: str = None
 
 
 class Project(CCIDictModel):
```

Comment:
I have Pydantic models in `feature/dependency-refactor` for the dependency list.